### DISCUSSION TITLE
immich community as an entry point, easily searchable knowledge database, for exchange

### DISCUSSION THREAD

**Discussion by aisbergde** [immich-app/immich#6257] [General]
I believe that a community and technical documentation should make it as easy as possible for users and administrators to use products. Perhaps because I myself work very intensively on the subject of technical documentation, docs-as-code, automated documentation, etc.

This does not mean that the documentation should be bloated to describe every possible tool. But helpful and further links would often be useful. Just to give the idea that there is something out there.

immich has quite good documentation, but it is still incomplete in many places. This makes it difficult to get started. You have to find a lot of things somewhere. On the other hand, the knowledge is there. It is often difficult to find and there is no easy entry point. Let's take the topic of CLIP.

Other projects use very good communication tools such as https://zulip.com/, which can be used excellently as a standardized place for synchronous and asynchronous communication between developers, users, for questions, FAQ etc., especially as a knowledge database. This is in addition to and independent of documentation, which should also remain "clean". Not everything has to be in the documentation, I completely agree with the developers. An additional standardized, easily searchable space that serves as a knowledge database makes it much easier to use products and software.

At immich, a lot of knowledge is distributed somewhere: some of it can be found on github in discussions or issues, although topics are often closed quickly. Then there's Discord. Then there's Reddit and all sorts of things. This means that the knowledge of individual users is available, but the exchange of knowledge between users is suboptimal. So everyone has to find individual solutions, even though others have already solved it several times.

I can't contribute much to development as I don't have the necessary skills. I could get involved in building a community.

**Comment by jrasm91** [comment]
IMHO we should take useful information from the other channels and add them into the docs website. There is some WIP for this in #5958. Other PRs are welcome that improve the official docs, etc.

**Reply by aisbergde** [reply]
This was also my first idea. But related feature requests are often closed, because "out of scope", #6254 maybe also some other requests.

So I thought to myself: If it is not desirable to add useful hints and links to the documentation, which some developers think are out of scope, then it would be good to have a less censored place where users can exchange ideas. Forums, Zulip, ... Where even "stupid" beginner questions or non-nerd questions can be asked and answered. The problem is that this usually splits up if it is not supported by the developers or a majority: Then there's one communication island here, another there. 

It would have to be something that can be found via a web search (which is problematic with Discord content) and that can provide public content without registration. 

I suggested zulip.com because I have had great experiences with it in various FOSS projects.

**Reply by bo0tzz** [reply]
These discussions right here are a community like that, no?

**Reply by aisbergde** [reply]
Github discussions could be used as a community. But that would require some other item-close-policies.

**Reply by jrasm91** [reply]
You opened that in the feature request category, which is why it was closed. I think that is fine TBH. You could open a separate post in show and tell or Question and Answer that asks how to connect to the database or show cases different UI tools you can use.

**Reply by aisbergde** [reply]
Yes, perhaps only the correct categories need to be used.
However, some feature requests are also closed because they are supposedly duplicates of other requests. However, this is sometimes not the case.

**Reply by jrasm91** [reply]
In those situations we can probably re-open the request.

**Comment by aisbergde** [comment]
You are right, for now, github discussions are fine, if we use it the right way, correctly distinguish between feature request, questions, and so on. And if we can request to reopen issues.
